Library/Homebrew/formulary.rb

```diff
@@ -237,18 +245,73 @@ def self.loader_for(ref)
     return NullLoader.new(ref)
   end
 
+  def self.find_with_priority(ref, is_installing, warn_all_ambiguity)
+    available_formulas = tap_paths(ref)
+    core_formula = core_path(ref)
+    if core_formula.file?
+      available_formulas[Tap.core_priority] = [] if available_formulas[50].nil?
+      available_formulas[Tap.core_priority] << core_formula
+    end
+
+    unless available_formulas.empty?
+      if warn_all_ambiguity
+        flat_formulas = available_formulas.values.flatten
+        if flat_formulas.length == 1
+          return flat_formulas.first
+        else
+          raise TapFormulaAmbiguityError.new(ref, flat_formulas)
+        end
+      else
+        if available_formulas.length > 1
+          opoo "Some formulae are excluded due to priority settings. Run brew info #{ref} to see them."
```

Comment:
Would be good to elaborate which formulae are excluded and give more information why.